app/controllers/conversations.js

```diff
@@ -216,24 +217,86 @@ class ConversationsController extends BaseController {
           validateParticipantsLimit,
         ]);
 
-        const participants = [];
-        for (let i = 0; i < addUsers.length; i++) {
+        const participants = await Promise.all(
+          addUsers.map(async (uId) => {
+            if (
+              !(await ConversationParticipant.findOne({
+                user_id: uId,
+                conversation_id: conversationId,
+              }))
+            ) {
+              return uId;
+            }
+          })
+        );
+
+        let currentUserParams;
+        const participantsInfo = (
+          await User.findAll(
+            { _id: { $in: [currentUserId, ...participants] } },
+            [
+              "login",
+              "first_name",
+              "last_name",
+              "email",
+              "phone",
+              "recent_activity",
+            ]
+          )
+        ).filter((obj) => {
+          if (obj._id.toString() === currentUserId) {
+            currentUserParams = obj;
+            return false;
+          }
+          return true;
+        });
+
+        const participantSavePromises = participantsInfo.map(async (u) => {
           const participant = new ConversationParticipant({
-            user_id: ObjectId(addUsers[i]),
+            user_id: ObjectId(u._id),
             conversation_id: ObjectId(conversationId),
           });
-          if (
-            !(await ConversationParticipant.findOne({
-              user_id: addUsers[i],
-              conversation_id: conversationId,
-            }))
-          ) {
-            await participant.save();
-            participants.push(participant.params._id);
-          }
-        }
+          await participant.save();
+
+          const messageInHistory = new Message({
+            id: currentUserId + Date.now(),
+            body: getUserName(u) + " has been added to the group",
+            cid: new ObjectId(conversationId),
+            deleted_for: [],
+            from: new ObjectId(currentUserId),
+            t: parseInt(Math.round(Date.now() / 1000)),
+            x: { type: "added_participant", user: u },
+          });
+          await messageInHistory.save();
+
+          const messageForDelivery = {
+            message: messageInHistory.visibleParams(),
+            push_message: {
+              title: `${getUserName(currentUserParams)} | ${conversation.name}`,
+              body: messageInHistory.params.body,
+              cid: messageInHistory.params.cid,
+            },
+          };
+
+          await PacketProcessor.deliverToUserOrUsers(
```

Comment:
I think there is a mistake.
We deliver this message to newly added users. But what about existing participants?